src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -144,13 +160,24 @@ public Object invokeMethod(String name, Object args) {
                     throw new GroovyRuntimeException(x);
                 }
             }
+            unreportedAmbiguousFunctions = new HashSet<>(0);
             for (StepDescriptor d : StepDescriptor.all()) {
-                functions.put(d.getFunctionName(), d);
+                String functionName = d.getFunctionName();
+                if (functions.containsKey(functionName)) {
+                    unreportedAmbiguousFunctions.add(functionName);
+                }
+                // The step descriptor with the highest value for Extension#ordinal() is used in the case of ambiguity.
+                functions.putIfAbsent(functionName, d);
+                stepClassNames.put(d.clazz.getName(), d);
             }
         }
-        final StepDescriptor sd = functions.get(name);
+        final StepDescriptor sd = functions.getOrDefault(name, stepClassNames.get(name));
         if (sd != null) {
-            return invokeStep(sd,args);
+            if (Util.isOverridden(DSL.class, getClass(), "invokeStep", StepDescriptor.class, Object.class) &&
+                    !Util.isOverridden(DSL.class, getClass(), "invokeStep", StepDescriptor.class, String.class, Object.class)) {
```

Comment:
Probably needless complexity. I know of no reason to subclass `DSL` except for the trick in `Snippetizer`.